stats/json/collector.go

```diff
@@ -94,44 +120,48 @@ func (c *Collector) HandleMetric(m *stats.Metric) {
 	}
 
 	c.seenMetrics = append(c.seenMetrics, m.Name)
-	env := WrapMetric(m)
-	row, err := json.Marshal(env)
+	err := c.encoder.Encode(WrapMetric(m))
 
-	if env == nil || err != nil {
-		logrus.WithField("filename", c.fname).Warning(
+	if err != nil {
+		logrus.WithField("filename", c.fname).WithField("error", err).Warning(
 			"JSON: Envelope is nil or Metric couldn't be marshalled to JSON")
 		return
 	}
-
-	row = append(row, '\n')
-	_, err = c.outfile.Write(row)
```

Comment:
I don't see where's the equivalent replacement for this code :confused:

Reply:
we are using an encoder that directly writes to the file instead of first marshaling and than writing to the file ... arguably the new line is required but apparently gets added as I checked the output and it looked the same. 

Now checking the [Encode](https://golang.org/pkg/encoding/json/#Encoder.Encode) documentation it adds a newline so this explains it :)